### PR TITLE
Remove hidden script access permission

### PR DESCRIPTION
**Pull request by dmcavoy**
Removes all the code related to HIDDEN_SCRIPT_ACCESS to make sure no one else adds that permission to an account.  As part of removing hidden script access we are removing the ability to assign alternative(experiment) units. @davidsbailey said we have been cleared to remove this feature and that it hasn't been used in years.  For UI tests that used hidden script access to assign the All The Things script, I moved to giving the levelbuilder permission to the user so that they can see and assign it.

## Links

[Jira](https://codedotorg.atlassian.net/browse/PLAT-1502)

## Testing story

- Updates to existing tests

### CODE REVIEW

File: dashboard/app/models/script.rb
Comment (dmcavoy) [inline dashboard/app/models/script.rb:348]: Can we update this to just check if someone can read the script???

Comment (davidsbailey) [reply dashboard/app/models/script.rb:348]: that would be amazing! :-) that would disable the course experiment stuff, which we are cleared to do. I'm not convinced we actually want admins to be able to view all scripts -- they should be using a levelbuilder account for that instead.

Comment (dmcavoy) [reply dashboard/app/models/script.rb:348]: Digging in here - Will update on progress :)

Comment (dmcavoy) [reply dashboard/app/models/script.rb:348]: I played with trying to move to can read a couple times here but it wasn't as simple as I had hoped. I decided to just get this working for now and look at the possibility of fixing this path later.

File: dashboard/test/ui/features/learning_platform/teacher_dashboard/progress_standards_view.feature
Comment (davidsbailey) [inline dashboard/test/ui/features/learning_platform/teacher_dashboard/progress_standards_view.feature:11]: nice adjustment!

**Review by davidsbailey — COMMENTED** [review]
I think this is the right strategy, but I'm a bit uneasy about doing this while /s/csp3-research-mxghyt is still getting 2000 hits per week. before we do this, I would suggest 
* moving that script to no longer be part of a csp unit group by removing the course experiment stuff from https://github.com/code-dot-org/code-dot-org/blob/eb983b71dfa90cc5f6b565d713f45887ae053e2d/dashboard/config/courses/csp-2017.course#L22
* make sure that script has reasonable values for published state, etc corresponding to no longer being in a unit group
* make sure the script id does not change in the process, so that progress is preserved

I would do that as a separate PR, wait a day to make sure we don't get zendesk tickets, and then proceed with the parts of this PR which get rid of the course experiment logic.

I think you can choose between holding off on this PR while you do that, or, adding back the course experiment logic into this PR if you want to unblock it. we have ui test coverage on course experiments, so I don't think you'd have to do a lot of manual testing if you go with the second option.

**Review by davidsbailey — APPROVED** [review]
hooray for removing dead features! LGTM once drone passes. please let me know if you'd like me to take another look.

**Pull request merged**
